core/src/mast/serialization/mod.rs

```diff
@@ -120,7 +120,9 @@ impl Deserializable for MastForest {
                 let node =
                     mast_node_info.try_into_mast_node(&mast_forest, &basic_block_data_decoder)?;
 
-                mast_forest.add_node(node);
+                mast_forest
+                    .add_node(node)
+                    .map_err(|e| DeserializationError::UnknownError(e.to_string()))?;
```

Comment:
Let's use `DeserializationError::InvalidValue` instead, and maybe with an error string that looks like `format!("failed to add node to MAST forest while deserializing: {}", e.to_string())`